internal/kmod/kmod.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2025 SUSE LLC
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kmod
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/suse/elemental/v3/pkg/cleanstack"
+	"github.com/suse/elemental/v3/pkg/deployment"
+	"github.com/suse/elemental/v3/pkg/sys"
+	"github.com/suse/elemental/v3/pkg/sys/vfs"
+)
+
+func ListKernelModules(s *sys.System) ([]string, error) {
+	d, err := deployment.Parse(s, "/")
+	if err != nil {
+		return nil, fmt.Errorf("parsing deployment information: %w", err)
+	} else if d == nil {
+		return nil, fmt.Errorf("deployment information not found")
+	}
+
+	var kernelModules []string
+
+	if d.Extensions != nil {
+		kernelModules = d.Extensions.KernelModules
+	}
+
+	// TODO (atanasdinov): Merge modules from deployment.yaml with ones coming from CLI args?
```

Comment:
In what sense? Are you thinking that the *originally* supplied kernel modules from `deployment.yaml` need to be appended to the list of modules now being supplied via sysext? Why wouldn't these match?